### ISSUE TITLE
Error while trying to get output

### ISSUE THREAD

**Issue by egorprnn** [tachibana-shin/fcanvas-next#3]
https://github.com/tachibana-shin/fcanvas-next/blob/master/src/Container.ts#L412

```js
        const horizontalPadding = 16;
        const verticalPadding = 12;
        const title = new Text({
            x: horizontalPadding,
            y: verticalPadding,
            text: 'test',
            fontSize: 28,
            fill: 'black',
            fontFamily: 'VK Sans Display'
        });

        const { width, height } = title['size']();

        const stageSize = {
            width: width + (horizontalPadding * 2),
            height: height + (verticalPadding * 2)
        };

        const stage = new Stage({
            container: 'title',
            ...stageSize
        });
        const titleLayer = new Layer();

        stage.add(titleLayer);

        const background = new Rect({
            fill: 'white',
            cornerRadius: 12,
            x: 0,
            y: 0,
            ...stageSize
        });

        titleLayer.add(background);
        // @ts-ignore
        titleLayer.add(title);

        return {
            url: stage.toDataURL(),
            ...stageSize
        };
```

![image](https://user-images.githubusercontent.com/42776347/165407645-55b84c6c-0cda-4e6b-8479-d809c965aaf7.png)

**Comment by tachibana-shin** [comment]
Thanks for reporting the bug!
However class `Stage` has officially removed `.toDataURL()` instead you can use `.toDataURL()` on class `Layer` like this
```diff
        const horizontalPadding = 16;
        const verticalPadding = 12;
        const title = new Text({
            x: horizontalPadding,
            y: verticalPadding,
            text: 'test',
            fontSize: 28,
            fill: 'black',
            fontFamily: 'VK Sans Display'
        });

        const { width, height } = title['size']();

        const stageSize = {
            width: width + (horizontalPadding * 2),
            height: height + (verticalPadding * 2)
        };

        const stage = new Stage({
            container: 'title',
            ...stageSize
        });
        const titleLayer = new Layer();

        stage.add(titleLayer);

        const background = new Rect({
            fill: 'white',
            cornerRadius: 12,
            x: 0,
            y: 0,
            ...stageSize
        });

        titleLayer.add(background);
        // @ts-ignore
        titleLayer.add(title);

        return {
-          url: stage.toDataURL(),
+         url:  titleLayer.toDataURL(),
            ...stageSize
        };
```

**Comment by tachibana-shin** [comment]
typo could not add `Text` to `Layer` (`tileLayer.add(title)`) fixed on version `fcanvas@^0.2.9` with some caching upgrade

**Comment by egorprnn** [comment]
@tachibana-shin oh, thx!

When using the method of the Layer constructor, an empty data url result is returned

![image](https://user-images.githubusercontent.com/42776347/165467507-b89e602f-c215-4bda-931a-689d2ab2a89a.png)

Output: `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASwAAACWCAYAAABkW7XSAAAAAXNSR0IArs4c6QAABGJJREFUeF7t1AEJAAAMAsHZv/RyPNwSyDncOQIECEQEFskpJgECBM5geQICBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAAYPlBwgQyAgYrExVghIgYLD8AAECGQGDlalKUAIEDJYfIEAgI2CwMlUJSoCAwfIDBAhkBAxWpipBCRAwWH6AAIGMgMHKVCUoAQIGyw8QIJARMFiZqgQlQMBg+QECBDICBitTlaAECBgsP0CAQEbAYGWqEpQAgQdWMQCX4yW9owAAAABJRU5ErkJggg==`

![image](https://user-images.githubusercontent.com/42776347/165470153-c6c04dd7-208b-45e2-bbdf-01d18f205672.png)

**Comment by tachibana-shin** [comment]
Hi,

This error fixed on version `fcanvas@^0.2.10`

**Comment by MrZillaGold** [comment]
@tachibana-shin i updated the package, but the problem is not solved :(

**Comment by egorprnn** [comment]
Now the layer is being drawn, but the size does not match.

Layer size: 232x52
Output: 300x150

`data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASwAAACWCAYAAABkW7XSAAAAAXNSR0IArs4c6QAAElVJREFUeF7tnXnMXNMbx5+DWH8ldlVqTexbaClqq732rVTtlNASGiqWdBFC1daqoHZqrX2LraJKimhTorUF/QNF7VskmF++R+/0vjP3zsx5+86cd+b93OQN6Tz3Pud87rnfe57nPGfGWYWjUCh0M7PeC/56Lfhvj0rn8BkEIFCRwDQz098MM5vpnPsUXrUTcHmmhUJhkJmNNrP1ar8clhCAQCCByWY2wjk3O/C8LmleJliFQmFVMxtvZgO6JBE6DYE4BMY454bHcd08XtsIVqFQGGFmI5un+bQUAi1FYJ5zrntL9aiDO1MUrEKhoNDvsw6+PpeDAATCCExwzg0JO6XrWKcF620zU2KdAwIQiEugj3NuetwmdE7vXrAKhcIwMxvbOZtIqyDQ5QhMcc7163K9rqHDrlAo/M/M5pjZWjXYYwIBCDSGwCjnHPnkEtYSrL5mNrUx9wAvEIBAjQSec871r9G2y5hJsIaa2bgu02M6CoHmIDDfOacSI44UAQnWJDMbCBUIQKDTEdjaOTer07UqYoMkWHPNrGfENuAaAhDIJjDYOTcROAsJSLAKAIEABDolgZHOuVGdsmWRGtUhgvXnn3/aueeea7fcckuxG48++qgddthhNXfr66+/tuOPP95efvllf85WW21lDz30kG200UY1XyPEUNc++uij7b777rNjjz22eOrnn39uxxxzjG277bZ2zTXX2NJLLx1yWWwh0JEEEKwSmnUTrBNPPNFuvPFGW2655Wq6gU899ZQdfPDBRdt6C9YHH3xgRx55pC2//PJ266232hZbbGE//fSTjRkzxq688kq77bbb7JRTTqmp7RhBoE4EEKx6C9Z5551nP/zwg7311lv2yCOP2GabbVb1Xv711192wQUX2EsvvWRnnXWWDRkypO4zrH/++ceuvfZa77f0kHBqtrj66qtXbTsGEKgjAQSr3oI1evRoW3/99W3QoEF288032+mnn171fn700Uc2YMAA22abbXyYtu+++9ZdsNSov//+2zSzu+mmm+yVV17x4admhppZrboqK8pVbxwG9SaAYDVCsCRWygOtueaadvvtt9uKK65Y8cbK5tRTTzXlvTSr2XnnnRsiWPUebVwfAotIAMFqhGApzNKfQsInnnjCevfWl5ZmH7/++qsPAz/++GN74IEH7KuvvkKwFnGUc3rLEECwGiFYl156qQ+1lAu67LLL7OKLLzbnsr/c9N1337UDDzzQh4JKdr/zzjtVBevff/+12bNn25133mmvvvqqzZw503drhx12sP79+/twdJ111sn0OWnSJP+5Qle1s/T45ZdfbOjQoXbPPff4jxTSXnfddbbMMstUfQqSa1c1XGAwbdo022mnncrM1T/lAB988EF74YUXTCHzSiutZDvuuKOfuYpr1mJGElqrbwcddJD9/vvv9vTTT9tjjz3mQ17lFsVlt91286GvZrJLLLFEbnNV8aJVU3HW6u306f99gYA477nnnjZw4EDbeOONc++tzp87d65fiX322WeL5yv033333e2kk06yTTfd1BZbbLGyNlS7TzohvTqdxVL+P/zwQ7v//vu9f42Tbt26Wb9+/fx9VRuWWmqpom/Z66V5xhln2B577JGZx6zULvHWi1opBo3lYcOGVeRbwzhBsBolWEmZgvzp4e/evfx7yTRAJAYjR460xx9/3A+kN954o6Jg6aGT0GhQ5B0alBKdCy+80A/Q9FHtQXjyySftuOOOM838YgjWvHnzvMDfcccduf3r1auXXX/99danT582YpEI1v777+9LRM455xx78cUXc69z6KGHev4SsdJDD58euhtuuKHIIutCZ555pn8pSVDTh8REZSFadU1YZp1/8skn2+WXX25rrLFG0H2qJljKT6pvaluef/GR77T46zy1W2NHfVP705/njZ9q59UgTlkmCFajBEurcKNGjfIDRg/NXnvtVXZD5s+f78VBNzsRtUqC9c033/g3o0RFCfLhw4f7mYQeFomfHnYNKD1oEraQAafG/fjjj/76WrXUA3/EEUe0a4aVN3uTj++//97XfWnmVDoryOrfIYcc4nOAYjRnzhwbO3asZyWRufvuu23XXXctck0E69tvv/X/pn7owZM/CYJmueKi2e9VV13lZ25ZK6LpmYLYpq+h2dDPP/9sb775phcz9aOUc6Xza21DtRdLNcF6/vnnvRiJ3fnnn28Sec2SxVEzrUsuucSPSwn/2Wef3Ub4K82UstpVy8wMwWongUYJlvxIfPbbbz8/9debKj391ucKU/SW1zQ6CRvzBCtdhrD99tv7MGWTTTbJpKAwU3kxhVWqsVJCPwlJKz0IyWcKYdZdd10/02tPSNgewVL/xOiiiy6ySv2TCKkcQ3YS64kTJ9pqq63mOSSCNWvWLC9omqUp7MkKxyV+ui9idMUVV/h7sPjii/vrJByyRDENXDMXidn777/vXx4Kx3UolB08eLB/kZSKavp8hYu6TwrX1B+94JIQdVEFq9rj8d577/k0RI8ePTIjgOTlMWXKFL/arVBcHLPa9dprr9kJJ5zgXwqVxmW1NmV8zgyrkYKlGYtKBJRIV25gvfUW/gCPHlCFdnfddVebxHyeYCUV6MpdTZ482fbee+/c+683ngaWZm8qkbj33nttlVVWafMwloqKZiWnnXaaF1XVYMlPIwUrpH8JV4XR6Ur9tGAprNHug7zcoWBohqFZpIpmleeRQKWvXSr2tTxwoedLEJRTlOimdzbUW7D++OMPHzLLpzgoL1d6JKKumXsivKXtyrKphVONNghWIwUryVEp+Zi3Baa09CFPsFRUKpEqFaC8G6+3t5LCEgIlnrXVJj17SAuW2qnKdrVT4qYwKWlHo2ZYof2TqCo5LFFSCLzkkksWZ1h6GCXqW265ZcXn4rvvvvOMlFBPwnbNPCRiyy67bLu2RoWer/BSM2C1V6G+Zo1596m0M9WS7vpcoaFESTNJjYmsQzNC1QFmHaWzpxkzZhQXbTSL1PgonYXVKEa1mCFYjRQs+UpWAffZZ582W3W0cnX44YeXFZfmCVbyZqtVQNK5onQOLevN/eWXX/rZ2Nprr23jx4/323UaLVihK4zJfVQYJpFVriaZYanoVTOmasWv6Qc+eaEk/db9UptWXnnlWh6sok3o+ek2pAuNQ3mU5gM/+eQTv2Oi0qJD0ujSl2m6w+n8VF7yXnlahcaVVlyDIC40RrAaLVhKYGrgaCAnb/1kkE6dOrVs+06jBUsDcty4cT48TYeazSJYaWFBsP4rEUmXphxwwAH+3irsTZemZIl1lqho/CrHp7+8o45bufia5EYLlvwllexKFiuESTYeK7lcukG60SHhF1984af4m2++eZt6q0YLVhISaqY3YcKEsnKMWt7QiWB19ZAwmdVvuOGGfqa51lrlP1eQ5LCUCsibYaVLFZKFkKyQUKFs1op0Lfesig2CFUOwkgdJewwlXg8//LDPv2QNlI5IuqdzIJWS7lqZ0sqcwkDlMXbZZZey0KbWELTWvEteWUOSdNfsQG2pln/KGujNmHTXLFurdR2ZdE/GkHJjKr1QPq70SHgrt5U1DtOhYHq1My/prut0ULFouqkIVgzBSr6NQUu+yrdoxSVr5VBta2RZw1FHHeWTraraVs1Ouuyi0TOsdFlDtRBDOTfVDkncJLiqHNfR6LKGpMRCdV0SdlXP64hd1pDMsHr27Fm2Oq32aeYkcUl2OpQKlsRKK7D65hHlrTTj1TjJK2tIl9F0sGghWDEESz6TJHviP726lW5TewtHdQ3VzmhFSBXOWhGqVDiaDFYVYSp3td1227VB02jBStqfFMaq0HHEiBG+tGKFFVbwhbEq+lShpoodtYWp9OFIBEvX0gKCijvrUTj622+/mUKjq6++2p555pmy4tNqhacqfVDtlYpgVTOWJdCLUtaQFABrL6vqp1TftcEGG/j7qxSA0hAKBZXTUjlLqWAlK4Pina7BqjSLrnTOIoSGCFYswUrKDPQQ6cirfm/U1pyEQ2nBYvLvMQRLvmvZmqPtRir0VBlGOpGcCJYKGPWgqfRBopZ3VNqao5mFzlW43J6tNbG35qQLY0v7L356YSlFoRKOtGClz8uaLeUJaV4IuQhipVMRrFiCld6qoxAsb39hNcFS+yttfq62KTe9XK5K+bwvGYwlWEn/ks3Pr7/+enHTrr71QrVoyvko3Ck90oKlfqrUQddR3lC1VnpphG5+TjYPpzc/KwRVOYXCpLzNy2pbpc3Pffv29f1QMrtem59VZ6a+qzhZbNR3jQ/tM1WOUN8Qoj5o6462L6Ur7/OS6JVmfgo1le5Q9FCtyr9GIUOw6iFYNcLHrM4ESgUrtIaqzs3j8uEEECwEK3zUNMsZCFaz3Kma24lgIVg1D5amM0Swmu6WVWswgoVgVRsjzfs5gtW89y6n5QgWgtVyg7rYIQSr5e4tgoVgtdygRrBa95YiWAhW645uetZyBBCsDMHSlwSVF/W03L2nQxBoOgKDnXMTm67VdWywfqp+kpkNrKMPLg0BCLSPwNbOuVntO7U1z5JgDTWzca3ZPXoFgaYlMN85x8+PZ4SEfc1satPeVhoOgdYk8Jxz7r9f9eAoEtAM639mNsfMyr/lDFAQgEAsAiTcM8j7n2MuFArDzGxsrDuDXwhAoA2BKc65fjApJ1D8/fhCofC2mfUCEgQgEJ1AH+fc9Oit6IQNSAuWfjTws07YRpoEga5EYIJzbkhX6nBIX4uCtSA0HGFmI0MugC0EINBhBOY557p32NVa8EJtBGuBaGkpdbyZZf+yZAtCoEsQ6AQExjjnhneCdnTqJpQJVtLaQqEwyMxGm9nC35fv1F2hcRBoSgKTzWyEc252U7a+wY3OFawFs61uZtZ7wZ8S8vr/Hg1uI+4g0EoEppmZ/maY2Uzn3Ket1Ll696WiYNXbOdeHAAQgEEIAwQqhhS0EIBCVAIIVFT/OIQCBEAIIVggtbCEAgagEEKyo+HEOAQiEEECwQmhhCwEIRCWAYEXFj3MIQCCEAIIVQgtbCEAgKgEEKyp+nEMAAiEEEKwQWthCAAJRCSBYUfHjHAIQCCGAYIXQwhYCEIhKAMGKih/nEIBACAEEK4QWthCAQFQCCFZU/DiHAARCCCBYIbSwhQAEohJAsKLixzkEIBBCAMEKoYUtBCAQlQCCFRU/ziEAgRACCFYILWwhAIGoBBCsqPhxDgEIhBBAsEJoYQsBCEQlgGBFxY9zCEAghACCFUILWwhAICoBBCsqfpxDAAIhBBCsEFrYQgACUQkgWFHx4xwCEAghgGCF0MIWAhCISgDBioof5xCAQAgBBCuEFrYQgEBUAghWVPw4hwAEQgggWCG0sIUABKISQLCi4sc5BCAQQgDBCqGFLQQgEJUAghUVP84hAIEQAghWCC1sIQCBqAQQrKj4cQ4BCIQQQLBCaGELAQhEJYBgRcWPcwhAIIQAghVCC1sIQCAqAQQrKn6cQwACIQQQrBBa2EIAAlEJIFhR8eMcAhAIIYBghdDCFgIQiEoAwYqKH+cQgEAIAQQrhBa2EIBAVAIIVlT8OIcABEIIIFghtLCFAASiEkCwouLHOQQgEEIAwQqhhS0EIBCVAIIVFT/OIQCBEAIIVggtbCEAgagEEKyo+HEOAQiEEECwQmhhCwEIRCWAYEXFj3MIQCCEAIIVQgtbCEAgKgEEKyp+nEMAAiEEEKwQWthCAAJRCSBYUfHjHAIQCCGAYIXQwhYCEIhKAMGKih/nEIBACAEEK4QWthCAQFQCCFZU/DiHAARCCCBYIbSwhQAEohJAsKLixzkEIBBCAMEKoYUtBCAQlQCCFRU/ziEAgRACCFYILWwhAIGoBBCsqPhxDgEIhBBAsEJoYQsBCEQlgGBFxY9zCEAghACCFUILWwhAICoBBCsqfpxDAAIhBBCsEFrYQgACUQkgWFHx4xwCEAghgGCF0MIWAhCISgDBioof5xCAQAgBBCuEFrYQgEBUAghWVPw4hwAEQgggWCG0sIUABKISQLCi4sc5BCAQQgDBCqGFLQQgEJUAghUVP84hAIEQAghWCC1sIQCBqAQQrKj4cQ4BCIQQQLBCaGELAQhEJYBgRcWPcwhAIIQAghVCC1sIQCAqAQQrKn6cQwACIQQQrBBa2EIAAlEJIFhR8eMcAhAIIYBghdDCFgIQiEoAwYqKH+cQgEAIAQQrhBa2EIBAVAIIVlT8OIcABEIIIFghtLCFAASiEkCwouLHOQQgEEIAwQqhhS0EIBCVwP8BZ4nPPJC0SNYAAAAASUVORK5CYII=`

**Comment by tachibana-shin** [comment]
This error fixed `fcanvas@^0.2.11`

**Comment by egorprnn** [comment]
Now it works as it should! Thanks for the wonderful library!
